modules/dnn/src/model.cpp

```diff
@@ -637,6 +637,296 @@ void DetectionModel::detect(InputArray frame, CV_OUT std::vector<int>& classIds,
+                box.y = (box.y - top) / resizeFactor;
+                box.width = (box.width - right) / resizeFactor;
+                box.height = (box.height - bottom)  / resizeFactor;
+            } else {
+                // Handle other cases or throw an error if the padding mode is unsupported
+                CV_Error(Error::StsNotImplemented, "Unsupported padding mode");
+            }
+            // convert from xmin, ymin, xmax, ymax -> xmin, ymin, width, hight
+            box.width = box.width - box.x;
+            box.height = box.height - box.y;
+        }
+    }
+};
+
+YOLODetectionModel::YOLODetectionModel(const String& model, const String& config)
+{
+    impl = makePtr<YOLODetectionModel_Impl>();
+    impl->initNet(readNet(model, config));
+}
+
+YOLODetectionModel::YOLODetectionModel(const String& onnx)
+{
+    impl = makePtr<YOLODetectionModel_Impl>();
+    impl->initNet(readNetFromONNX(onnx));
+}
+
+YOLODetectionModel::YOLODetectionModel()
+{
+    impl = std::static_pointer_cast<Model::Impl>(makePtr<YOLODetectionModel_Impl>());
+}
+
+void YOLODetectionModel::detect(InputArray frame, CV_OUT std::vector<int>& classIds,
+                            CV_OUT std::vector<float>& confidences, CV_OUT std::vector<Rect>& boxes,
+                            float confThreshold, float nmsThreshold){
+
+    CV_Assert(impl != nullptr && impl.dynamicCast<YOLODetectionModel_Impl>() != nullptr); // remove once default constructor is removed
+
+    std::vector<Mat> detections;
+    impl.dynamicCast<YOLODetectionModel_Impl>()->processFrame(frame, detections);
+
+    postProccess(
+        detections,
+        boxes,
+        confidences,
+        classIds,
+        impl->size,
+        confThreshold,
+        nmsThreshold,
+        getNmsAcrossClasses()
+        );
+
+    impl.dynamicCast<YOLODetectionModel_Impl>()->boxRescale(boxes);
+}
+
+void YOLODetectionModel::postProccess(
+    std::vector<Mat>& detections,
+    CV_OUT std::vector<Rect>& keep_boxes,
+    CV_OUT std::vector<float>& keep_confidences,
+    CV_OUT std::vector<int>& keep_classIds,
+    Size inputImgSize,
+    const float confThreshold,
+    const float nmsThreshold,
+    const bool nmsAcrossClasses
+    ){
+
+    bool yolov8 = false;
+    bool darknet = false;
+
+    if (detections[0].dims == 2){
+        darknet = true;
+    }
+
+    if (!darknet && detections[0].size[1] < detections[0].size[2]) {
+        yolov8 = true;  // Set the correct flag based on tensor shape
+    }
+
+    // Retrieve
+    std::vector<int> classIds;
+    std::vector<float> confidences;
+    std::vector<Rect> boxes;
+
+    if (yolov8){
+        for(auto & detection : detections){
+            cv::transposeND(detection, {0, 2, 1}, detection);
+        }
+    }
+
+    // each row is [cx, cy, w, h, conf_obj, conf_class1, ..., conf_class80]
+    for (auto preds : detections)
+    {
+        if (!darknet)
+            preds = preds.reshape(1, preds.size[1]);
```

Comment:
This step is valid for every model. So can we remove the `if` condition?

Reply:
Not valid. For yolov5 and larger model not valid

Reply:
Can you please share the shapes for them?

Reply:
Can you please share the shapes for them?